### PR TITLE
feature/dao-dashboard-contribution-percent-member (PRO-478)

### PR DESCRIPTION
**Pull request by jonathanprozzi**
## Linear Ticket

- [PRO-478](https://linear.app/govrn/issue/PRO-478/[pie]-contribution-percentage-per-member)
- [PRO-569](https://linear.app/govrn/issue/PRO-569/create-usedaocontributionsbyuser-hook)

## Description

- Adds the Contribution Members Pie Chart and a `useDaoContributionCountByUserInRange` hook
- Data flow is the same as the other DAO Dashboard charts: DaoDashboardShell -> ContributionMembersPieShell -> ContributionMembersPie so we can make any data changes in the shells and pass into the main UI component
- Minor layout adjustment to the DaoDashboardShell since this is in, but this'll need to be tweaked overall. Mobile layout is looking solid -- will need to refine the desktop

## Screencaptures

https://user-images.githubusercontent.com/9438776/199396083-50798d08-4cca-4194-aafc-302b7e92d2f1.mp4

### CODE REVIEW

File: apps/protocol-frontend/src/components/ContributionMembersPie.tsx [hunk 1]
```diff
@@ -0,0 +1,138 @@
+import { Flex, Heading, Text, useBreakpointValue } from '@chakra-ui/react';
+import { ResponsivePie } from '@nivo/pie';
+import { GovrnSpinner } from '@govrn/protocol-ui';
+
+type ContributionMembersCount = {
+  count: number;
+  display_name?: string | null | undefined;
+  user_id?: number | null | undefined;
```
Comment (alexkeating) [inline apps/protocol-frontend/src/components/ContributionMembersPie.tsx:8]: I think undefined is redundant with the ?.

File: apps/protocol-frontend/src/components/ContributionMembersPie.tsx [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+};
+interface ContributionTypesPieProps {
+  contributionMembersData: ContributionMembersCount[] | undefined;
```
Comment (alexkeating) [inline apps/protocol-frontend/src/components/ContributionMembersPie.tsx:11]: Same here use the ?

File: apps/protocol-frontend/src/components/ContributionMembersPie.tsx [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+const brandColorMap = [
+  '#ffb4e2',
+  '#fb84ce',
+  '#f854ba',
+  '#f526a6',
+  '#db0f8d',
+  '#ab076d',
+  '#76024e',
+];
```
Comment (alexkeating) [inline apps/protocol-frontend/src/components/ContributionMembersPie.tsx:25]: This brand color map is in multiple spots we should move it to a constants file

File: apps/protocol-frontend/src/components/ContributionMembersPie.tsx [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+  isFetching,
+  isLoading,
+  isError,
+}: ContributionTypesPieProps) => {
+  const isMobile = useBreakpointValue({ base: true, lg: false });
+
+  const contributionsDataMap = contributionMembersData?.map(contribution => {
+    return {
+      id:
+        contribution.display_name !== null
```
Comment (alexkeating) [inline apps/protocol-frontend/src/components/ContributionMembersPie.tsx:38]: Could display_name be undefined as well?

File: apps/protocol-frontend/src/components/ContributionMembersPie.tsx [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+        background="white"
+        boxShadow="sm"
+        borderRadius={{ base: 'none', md: 'md' }}
+      >
+        <Heading as="h3" size="md" color="gray.800" fontWeight="normal">
+          Contributions By Member
+        </Heading>
+        {contributionsDataMap?.length !== 0 ? (
+          <ResponsivePie
+            data={contributionsDataMap ? contributionsDataMap : []}
```
Comment (alexkeating) [inline apps/protocol-frontend/src/components/ContributionMembersPie.tsx:76]: Probably more succinct with contributionsDataMap || []

**Review by alexkeating — CHANGES_REQUESTED** [review]

File: apps/protocol-frontend/src/components/ContributionMembersPie.tsx (hunk 4 above)
Comment (jonathanprozzi) [reply apps/protocol-frontend/src/components/ContributionMembersPie.tsx]: I'll add a protection for that too -- I think they're null if not set, but good to be safe

**Review by alexkeating — APPROVED** [review]

**Pull request merged**